packages/systemd/9013-systemd-networkd-Conditionalize-hostnamed-timezoned-DBUS.patch

```diff
@@ -0,0 +1,121 @@
+         return 0;
+ }
+ 
++#if ENABLE_HOSTNAMED
+ int manager_request_product_uuid(Manager *m) {
+         static bool bus_method_is_called = false;
+         int r;
+@@ -207,6 +208,7 @@ int manager_request_product_uuid(Manager *m) {
+ 
+         return 0;
+ }
++#endif
+ 
+ int dhcp_configure_duid(Link *link, const DUID *duid) {
+         Manager *m;
+diff --git a/src/network/networkd-dhcp-common.h b/src/network/networkd-dhcp-common.h
+index c19bc10..9d13cde 100644
+--- a/src/network/networkd-dhcp-common.h
++++ b/src/network/networkd-dhcp-common.h
+@@ -72,7 +72,13 @@ static inline const DUID *link_get_dhcp6_duid(Link *link) {
+ }
+ 
+ int dhcp_configure_duid(Link *link, const DUID *duid);
++#if ENABLE_HOSTNAMED
+ int manager_request_product_uuid(Manager *m);
++#else
++static inline int manager_request_product_uuid(_unused_ Manager *m) {
++        return 0;
++}
++#endif
+ 
+ bool address_is_filtered(int family, const union in_addr_union *address, uint8_t prefixlen, Set *allow_list, Set *deny_list);
+ static inline bool in4_address_is_filtered(const struct in_addr *address, Set *allow_list, Set *deny_list) {
+diff --git a/src/network/networkd-manager.c b/src/network/networkd-manager.c
+index ff2770f..a225526 100644
+--- a/src/network/networkd-manager.c
++++ b/src/network/networkd-manager.c
+@@ -845,6 +845,7 @@ static int set_hostname_handler(sd_bus_message *m, void *userdata, sd_bus_error
+         return 1;
+ }
+ 
++#if ENABLE_HOSTNAMED
+ int manager_set_hostname(Manager *m, const char *hostname) {
+         int r;
+ 
+@@ -876,6 +877,7 @@ int manager_set_hostname(Manager *m, const char *hostname) {
+ 
+         return 0;
+ }
++#endif
+ 
+ static int set_timezone_handler(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
+         const sd_bus_error *e;
+@@ -892,6 +894,7 @@ static int set_timezone_handler(sd_bus_message *m, void *userdata, sd_bus_error
+         return 1;
+ }
+ 
++#if ENABLE_HOSTNAMED
+ int manager_set_timezone(Manager *m, const char *tz) {
+         int r;
+ 
+@@ -925,3 +928,4 @@ int manager_set_timezone(Manager *m, const char *tz) {
+ 
+         return 0;
+ }
++#endif
+diff --git a/src/network/networkd-manager.h b/src/network/networkd-manager.h
+index 86de529..b80c6c5 100644
+--- a/src/network/networkd-manager.h
++++ b/src/network/networkd-manager.h
+@@ -110,7 +110,20 @@ bool manager_should_reload(Manager *m);
+ 
+ int manager_enumerate(Manager *m);
+ 
++#if ENABLE_HOSTNAMED
+ int manager_set_hostname(Manager *m, const char *hostname);
++#else
++static inline int manager_set_hostname(_unused_ Manager *m, _unused_ const char *hostname) {
++        return 0;
++}
++#endif /* ENABLE_HOSTNAMED */
++
++#if ENABLE_HOSTNAMED
+ int manager_set_timezone(Manager *m, const char *timezone);
++#else
++static inline int manager_set_timezone(_unused_ Manager *m, _unused_ const char *timezone) {
++        return 0;
++}
++#endif /* ENABLE_HOSTNAMED */
```

Comment:
This doesn't look right - `hostnamed` is required for `manager_set_timezone`?

Reply:
Fixed!